yt_dlp/extractor/mediastream.py

```diff
@@ -118,7 +122,29 @@ def _real_extract(self, url):
         formats, subtitles = [], {}
         for video_format in player_config['src']:
             if video_format == 'hls':
-                fmts, subs = self._extract_m3u8_formats_and_subtitles(player_config['src'][video_format], video_id)
+                params = {}
+
+                uid = self._search_regex(r'window\.MDSTRMUID\s*=\s*["\']([^"\']+)["\'];', webpage, 'uid', default=None)
+                if uid:
+                    params['uid'] = uid
+
+                sid = self._search_regex(r'window\.MDSTRMSID\s*=\s*["\']([^"\']+)["\'];', webpage, 'sid', default=None)
+                if sid:
+                    params['sid'] = sid
+
+                pid = self._search_regex(r'window\.MDSTRMPID\s*=\s*["\']([^"\']+)["\'];', webpage, 'pid', default=None)
+                if pid:
+                    params['pid'] = pid
+
+                version = self._search_regex(r'window\.VERSION\s*=\s*["\']([^"\']+)["\'];', webpage, 'version', default=None)
+                if version:
+                    params['at'] = 'web-app'
+                    params['av'] = version
+
+                if access_token := parse_qs(url).get('access_token', [None])[0]:
+                    params['access_token'] = access_token
+
+                fmts, subs = self._extract_m3u8_formats_and_subtitles(update_url_query(player_config['src'][video_format], params), video_id)
```

Comment:
Sorry, I forgot the line break
```suggestion
                fmts, subs = self._extract_m3u8_formats_and_subtitles(
                    update_url_query(player_config['src'][video_format], params), video_id)
```